### PR TITLE
asl: new port (version 20230818)

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.6.8 21G725 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
@herbygillot, @mascguy may I ask you to review and merge it?

It should unblock fix for https://trac.macports.org/ticket/68031

File: math/asl/Portfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+github.setup                    ampl asl 94efb9e220c963bddee3d4500c28e86eb560f858
+version                         20230818
+revision                        0
+
+categories                      math
+maintainers                     {@catap korins.ky:kirill} openmaintainer
+
+# It is similar for so-called CWI license which used for Python between 0.9.0 and 1.2
+license                         Unknown
```
Comment (mascguy) [inline math/asl/Portfile:16]: Kirill, is the license Permissive-ish, such that we can redistribute binaries? Or should we avoid distribution, just-in-case?

CC: @jmroot

Comment (catap) [reply math/asl/Portfile:16]: @mascguy as far as I see we can redistirbute it without any reason.

See the license here: https://github.com/ampl/asl/blob/master/src/solvers/asl.h#L1-L17

Comment (jmroot) [reply math/asl/Portfile:16]: A lot of the files don't have any license header, and there's no blanket license specified anywhere, which makes things questionable at best. https://github.com/ampl/asl/issues/7

Many of the files that do have a license are an MIT variant, but you'd have to go through all of them to make sure there aren't any other licenses in there.

Comment (mascguy) [reply math/asl/Portfile:16]: > A lot of the files don't have any license header, and there's no blanket license specified anywhere, which makes things questionable at best. [ampl/asl#7](https://github.com/ampl/asl/issues/7)
> 
> Many of the files that do have a license are an MIT variant, but you'd have to go through all of them to make sure there aren't any other licenses in there.

So would `Restrictive` be the safest at this point? Or does `Unknown` provide the same net effect?

Comment (catap) [reply math/asl/Portfile:16]: To be on safe side I've added `Restrictive` and keep `Unknown`.

**Review by mascguy — APPROVED** [review]

**Pull request merged**